components/rich_text_editor/rich_text_editor.vue

```diff
@@ -0,0 +1,250 @@
+<template>
+  <editor-content
+    :editor="editor"
+    data-qa="dt-rich-text-editor"
+  />
+</template>
+
+<script>
+import { Editor, EditorContent } from '@tiptap/vue-2';
+import Document from '@tiptap/extension-document';
+import Paragraph from '@tiptap/extension-paragraph';
+import Text from '@tiptap/extension-text';
+import {
+  RICH_TEXT_EDITOR_OUTPUT_FORMATS,
+  RICH_TEXT_EDITOR_AUTOFOCUS_TYPES,
+} from './rich_text_editor_constants';
+
+export default {
+  name: 'DtRichTextEditor',
+
+  components: {
+    EditorContent,
+  },
+
+  props: {
+    /**
+     * Value of the input
+     */
+    value: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input is editable
+     */
+    editable: {
+      type: Boolean,
+      default: true,
+    },
+
+    /**
+     * Descriptive label for the input element
+     */
+    inputAriaLabel: {
+      type: String,
+      required: true,
+    },
+
+    /**
+     * Additional class name for the input element. Only accepts a String value
+     * because this is passed to the editor via options. For multiple classes,
+     * join them into one string, e.g. "d-p8 d-hmx96"
+     */
+    inputClass: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input should receive focus after the component has been
+     * mounted. Either one of `start`, `end`, `all` or a Boolean or a Number.
+     * - `start`  Sets the focus to the beginning of the input
+     * - `end`    Sets the focus to the end of the input
+     * - `all`    Selects the whole contents of the input
+     * - `Number` Sets the focus to a specific position in the input
+     * - `true`   Defaults to `start`
+     * - `false`  Disables autofocus
+     * @values true, false, start, end, all, number
+     */
+    autofocus: {
+      type: [Boolean, String, Number],
+      default: false,
+      validator (autofocus) {
+        if (typeof autofocus === 'string') {
+          return RICH_TEXT_EDITOR_AUTOFOCUS_TYPES.includes(autofocus);
+        }
+        return true;
+      },
+    },
+
+    /**
+     * The output format that the editor uses when emitting the "@input" event.
+     * One of `text`, `json`, `html`. See https://tiptap.dev/guide/output for
+     * examples.
+     * @values text, json, html
+     */
+    outputFormat: {
+      type: String,
+      default: 'text',
+      validator (outputFormat) {
+        return RICH_TEXT_EDITOR_OUTPUT_FORMATS.includes(outputFormat);
+      },
+    },
+  },
+
+  emits: [
+    /**
+     * Editor input event
+     * @event input
+     * @type {String|JSON}
+     */
+    'input',
+
+    /**
+     * Event to sync the value with the parent
+     * @event input
+     * @type {String|JSON}
+     */
+    'update:value',
```

Comment:
Will a `.sync` actually work here? the output is completely different than what's being passed in right?

Reply:
Hmm see you already thought about this in the value watch

```
if (newValue === this.getOutput()) {
  // The new value came from this component and was passed back down
  // through the parent, so don't do anything here.
  return;
}
```

Not sure I love this, since it's not really how v-model / sync is expected to work and at this point you're not really "syncing" anything

Maybe we have `input` / `update:value` events ALWAYS output as text, and have a different event for the formatted output

Reply:
> Will a `.sync` actually work here? the output is completely different than what's being passed in right?

This will expect that the input is in the format that the output is. `v-model` works here and AFAIK it's the same thing as .sync except defaulting to the `value` prop. Not sure how useful it really is to sync the value from the parent except when setting/resetting the initial value for this.

> Not sure I love this, since it's not really how v-model / sync is expected to work and at this point you're not really "syncing" anything

Hmm, maybe the comment is unclear here, but what this early return is doing is just stopping the value from the editor from being passed straight back to it since it would reset the cursor position.
- Editor -> `this.$emit('update:value', '<p>hello</p>')`
- Parent -> local value changes from `<p>hell</p>` to `<p>hello</p>` and as the value here has changed it gets passed right down to the Editor again
- Editor receives `<p>hello</p>`, which matches what the value already is, so there's no need to set it again

Or am I misunderstanding something? I was going back and forth whether to include this in the first place, since the Editor really should be the source of truth for the value. So instead of `value` we could have a prop `initialValue` that makes it clear that it resets the value of the Editor and it wouldn't support v-model, but to get the new value you'd have `@output` or whatever

Reply:
alright so I think what you're saying is whatever format is output, the library input field will also take as input?

ex if you pass `<p>hello</p>` as the input value it only displays as `hello` in the input itself, not `<p>hello</p>`

Reply:
Right, the editor will render any HTML passed to it as a value, but you can't actually type HTML tags in the text as they'll be escaped. The value you pass to it is not actually limited to the selected output format, but for practical reasons they should probably match. Like I can choose to output JSON and pass HTML as the initial value and the editor will not complain.

Reply:
alright, I think this is okay then.